src/main/java/org/elasql/perf/tpart/metric/TransactionMetricRecorder.java

```diff
@@ -30,33 +30,57 @@
 	private static final String TRANSACTION_ID_COLUMN = "Transaction ID";
 	private static final String MASTER_ID_COLUMN = "Is Master";
 	private static final long TIME_TO_FLUSH = 10; // in seconds
+	private static final boolean ENABLE_CPU_TIMER = TransactionProfiler.ENABLE_CPU_TIMER;
+	private static final boolean ENABLE_DISKIO_COUNTER = TransactionProfiler.ENABLE_DISKIO_COUNTER;
 	
 	private static class TransactionMetrics {
 		long txNum;
 		boolean isMaster;
 		List<String> metricNames;
 		Map<String, Long> latencies;
+		Map<String, Long> cpuTimes;
+		Map<String, Long> diskioCounts;
 		
-		TransactionMetrics(long txNum, String role, Timer timer) {
+		TransactionMetrics(long txNum, String role, TransactionProfiler profiler) {
 			this.txNum = txNum;
 			this.isMaster = role.equals("Master");
 			
 			metricNames = new ArrayList<String>();
 			latencies = new HashMap<String, Long>();
-			for (Object component : timer.getComponents()) {
+			cpuTimes = new HashMap<String, Long>();
+			diskioCounts = new HashMap<String, Long>();
+			for (Object component : profiler.getComponents()) {
 				String metricName = component.toString();
 				metricNames.add(metricName);
-				latencies.put(metricName, timer.getComponentTime(component));
+				latencies.put(metricName, profiler.getComponentTime(component));
+				if (ENABLE_CPU_TIMER)
+					cpuTimes.put(metricName, profiler.getComponentCpuTime(component));
+				if (ENABLE_DISKIO_COUNTER)
+					diskioCounts.put(metricName, profiler.getComponentIOCount(component));
 			}
 		}
 	}
 	
-	private static class LatencyRow implements CsvRow, Comparable<LatencyRow> {
+	private AtomicBoolean isRecording = new AtomicBoolean(false);
+	private BlockingQueue<TransactionMetrics> queue
+		= new ArrayBlockingQueue<TransactionMetrics>(100000);
+	
+	// Header
+	private int serverId;
+	private List<Object> metricNames = new ArrayList<Object>();
+	private Map<Object, Integer> metricNameToPos = new HashMap<Object, Integer>();
+	
+	// Data
+	private List<LongValueRow> latencyRows = new ArrayList<LongValueRow>();
+	private List<LongValueRow> cpuTimeRows = new ArrayList<LongValueRow>();
+	private List<LongValueRow> diskioCountRows = new ArrayList<LongValueRow>();
+	
+	private static class LongValueRow implements CsvRow, Comparable<LongValueRow> {
```

Comment:
It would be better to put a static class with other static members (e.g., static classes, static variables...).